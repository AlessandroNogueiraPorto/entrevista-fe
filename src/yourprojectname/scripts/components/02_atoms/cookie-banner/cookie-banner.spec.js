/* eslint-env jest */
import Vue from 'vue'
import cookieBanner from './cookie-banner.vue'
import { createRenderer } from 'vue-server-renderer'

describe('cookie-banner', () => {
  let SUT
  let vm
  let renderer

  beforeEach(() => {
    SUT = Vue.extend(cookieBanner)
    vm = new SUT({
      propsData: {
        buttonText: 'Dismiss'
      }
    })
    renderer = createRenderer()
  })

  it('should be a component', () => {
    expect(cookieBanner).toBeDefined()

    renderer.renderToString(vm, (err, str) => {
      expect(err).toBeNull()
      expect(str).toMatchSnapshot()
    })
  })

  it('should be open by default', () => {
    expect(vm.showBanner).toBe(true)

    renderer.renderToString(vm, (err, str) => {
      expect(err).toBeNull()
      expect(str).toMatchSnapshot()
    })
  })
})
