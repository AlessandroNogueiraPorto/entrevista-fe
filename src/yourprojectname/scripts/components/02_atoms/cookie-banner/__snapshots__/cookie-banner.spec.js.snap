// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cookie-banner should be a component 1`] = `

<div data-server-rendered="true"
     class="cookie-banner"
>
  <div class="cookie-banner__close">
    <svg class="icon is-large">
      <use xmlns:xlink="http://www.w3.org/1999/xlink"
           xlink:href="#close"
      >
      </use>
    </svg>
  </div>
  <div class="cookie-banner__content">
    <button class="cookie-submit-button">
      Dismiss
    </button>
  </div>
</div>

`;

exports[`cookie-banner should be open by default 1`] = `

<div data-server-rendered="true"
     class="cookie-banner"
>
  <div class="cookie-banner__close">
    <svg class="icon is-large">
      <use xmlns:xlink="http://www.w3.org/1999/xlink"
           xlink:href="#close"
      >
      </use>
    </svg>
  </div>
  <div class="cookie-banner__content">
    <button class="cookie-submit-button">
      Dismiss
    </button>
  </div>
</div>

`;
